workflow/artifacts/s3/s3.go

```diff
@@ -85,11 +88,14 @@ func (s3Driver *S3ArtifactDriver) Save(path string, outputArtifact *wfv1.Artifac
 			}
 			if isDir {
 				if err = s3cli.PutDirectory(outputArtifact.S3.Bucket, outputArtifact.S3.Key, path); err != nil {
+					log.Warnf("Failed to put directory: %v", err)
+					return false, nil
+				}
+			} else {
+				if err = s3cli.PutFile(outputArtifact.S3.Bucket, outputArtifact.S3.Key, path); err != nil {
```

Comment:
As discussed on Slack, would it be worth adding an info log to stat the file size for debug purposes, in the event that we see a discrepancy between the file on disk and the file in the S3 bucket?